frontend/controller/app/identity.js

```diff
@@ -0,0 +1,379 @@
+          // Vue.delete not needed as the entire object will replace the state
+          delete state[k]
+        })
+      }
+      // Augment state with Chelonia state
+      Object.keys(cheloniaState.contracts).forEach(k => {
+        if (cheloniaState[k]) {
+          // Vue.set not needed as the entire object will replace the state
+          state[k] = cheloniaState[k]
+        }
+      })
+      state.contracts = cheloniaState.contracts
+      if (cheloniaState.namespaceLookups) {
+        state.namespaceLookups = cheloniaState.namespaceLookups
+      }
+      // End exclude contracts
+      sbp('state/vuex/postUpgradeVerification', state)
+      sbp('state/vuex/replace', state)
+    } else {
+      // Else, if `state` was not given, just sync add contracts from Chelonia
+      // to the current Vuex state
+      const state = vuexState
+      // Exclude contracts from the state
+      if (state.contracts) {
+        Object.keys(state.contracts).forEach(k => {
+          Vue.delete(state, k)
+        })
+      }
+      Object.keys(cheloniaState.contracts).forEach(k => {
+        if (cheloniaState[k]) {
+          Vue.set(state, k, cheloniaState[k])
+        }
+      })
+      Vue.set(state, 'contracts', cheloniaState.contracts)
+      if (cheloniaState.namespaceLookups) {
+        Vue.set(state, 'namespaceLookups', cheloniaState.namespaceLookups)
+      }
+    // End exclude contracts
+    }
+
+    if (encryptionParams) {
+      sbp('state/vuex/commit', 'login', { identityContractID, encryptionParams })
+    }
+
+    // NOTE: users could notice that they leave the group by someone
+    // else when they log in
+    const currentState = sbp('state/vuex/state')
+    if (!currentState.currentGroupId) {
+      const gId = Object.keys(currentState.contracts)
+        .find(cID => has(currentState[identityContractID].groups, cID))
+
+      if (gId) {
+        sbp('gi.app/group/switch', gId)
+      }
+    }
+
+    // Whenever there's an active session, the encrypted save state should be
+    // removed, as it is only used for recovering the state when logging in
+    sbp('gi.db/settings/deleteEncrypted', identityContractID).catch(e => {
+      console.error('Error deleting encrypted settings after login')
+    })
+
+    // TODO: [SW] It may make more sense to load persistent actions in
+    // actions in the SW instead of on each tab
+    const databaseKey = `chelonia/persistentActions/${identityContractID}`
+    sbp('chelonia.persistentActions/configure', { databaseKey })
+    await sbp('chelonia.persistentActions/load')
+
+    sbp('okTurtles.events/emit', LOGIN_COMPLETE, { identityContractID })
+  } catch (e) {
+    sbp('okTurtles.events/emit', LOGIN_ERROR, { identityContractID, error: e })
+  }
+})
+
+sbp('okTurtles.events/on', LOGOUT, (a) => {
+  // TODO: [SW] It may make more sense to load persistent actions in
+  // actions in the SW instead of on each tab
+  sbp('chelonia.persistentActions/unload')
+})
+
```

Comment:
We are not using persistent actions at all right now, so I think all persistent action related code should be commented out (including the imports, to reduce bundle size).